assets/js/darkmode.js

```diff
@@ -1,18 +1,31 @@
 const userPref = window.matchMedia('(prefers-color-scheme: light)').matches ? 'light' : 'dark'
 const currentTheme = localStorage.getItem('theme') ?? userPref
+const syntaxTheme = document.querySelector("#theme-link");
+
+
+{{ $darkSyntax := resources.Get "styles/_dark_syntax.scss" | resources.ToCSS (dict "outputStyle" "compressed") | resources.Fingerprint "md5" | resources.Minify  }}
+{{ $lightSyntax := resources.Get "styles/_light_syntax.scss" | resources.ToCSS (dict "outputStyle" "compressed") | resources.Fingerprint "md5" | resources.Minify  }}
 
 if (currentTheme) {
   document.documentElement.setAttribute('saved-theme', currentTheme);
+  if (currentTheme === 'dark') {
+    syntaxTheme.href = '{{ $darkSyntax.Permalink }}';
```

Comment:
woah I didn't know you could just hot-swap stylesheets like this, does this work in all the browsers you've tested with?

Reply:
nit: this might also be cleaner as a ternary

Reply:
Works on Chrome and Firefox. On mobile I can't say that this **specific implementation** works, as I haven't deployed it, but in an **almost identical implementation** I swapped my `giscus` comments theme in quartz, successfully across every device I tested with. I can't say anything about IE though lol.

[Here's a nice resource I found that details all darkmode implementation methods](https://css-tricks.com/a-complete-guide-to-dark-mode-on-the-web/), It mentions browser support but doesn't speak of any limitations of this specific method.

Reply:
Ternary operator added, and there seems to be no concern of browser incompatibilities with the hot-swap of style sheets.